ledger/acctonline_test.go

```diff
@@ -279,3 +283,138 @@ func TestAcctOnline(t *testing.T) {
+
+	accts := []map[basics.Address]basics.AccountData{ledgertesting.RandomAccounts(20, true)}
+
+	pooldata := basics.AccountData{}
+	pooldata.MicroAlgos.Raw = 1000 * 1000 * 1000 * 1000
+	pooldata.Status = basics.NotParticipating
+	accts[0][testPoolAddr] = pooldata
+
+	sinkdata := basics.AccountData{}
+	sinkdata.MicroAlgos.Raw = 1000 * 1000 * 1000 * 1000
+	sinkdata.Status = basics.NotParticipating
+	accts[0][testSinkAddr] = sinkdata
+
+	ml := makeMockLedgerForTracker(t, true, 10, protocol.ConsensusCurrentVersion, accts)
+	defer ml.Close()
+
+	conf := config.GetDefaultLocal()
+	au, ao := newAcctUpdates(t, ml, conf, ".")
+	defer au.close()
+	defer ao.close()
+
+	// cover 10 genesis blocks
+	rewardLevel := uint64(0)
+	for i := 1; i < 10; i++ {
+		accts = append(accts, accts[0])
+	}
+
+	// lastCreatableID stores asset or app max used index to get rid of conflicts
+	lastCreatableID := crypto.RandUint64() % 512
+	knownCreatables := make(map[basics.CreatableIndex]bool)
+
+	allTotals := make(map[basics.Round]ledgercore.AccountTotals)
+
+	start := basics.Round(10)
+	end := basics.Round(2*proto.MaxBalLookback + 15)
+	for i := start; i < end; i++ {
+		consensusVersion := protocol.ConsensusCurrentVersion
+		if i > basics.Round(proto.MaxBalLookback) {
+			consensusVersion = protocol.ConsensusFuture
+		}
+		rewardLevelDelta := crypto.RandUint64() % 3
+		rewardLevel += rewardLevelDelta
+		var updates ledgercore.AccountDeltas
+		var totals map[basics.Address]ledgercore.AccountData
+		base := accts[i-1]
+		updates, totals, lastCreatableID = ledgertesting.RandomDeltasBalancedFull(1, base, rewardLevel, lastCreatableID)
+		prevRound, prevTotals, err := au.LatestTotals()
+		require.Equal(t, i-1, prevRound)
+		require.NoError(t, err)
+
+		newPool := totals[testPoolAddr]
+		newPool.MicroAlgos.Raw -= prevTotals.RewardUnits() * rewardLevelDelta
+		updates.Upsert(testPoolAddr, newPool)
+		totals[testPoolAddr] = newPool
+		newAccts := applyPartialDeltas(base, updates)
+
+		blk := bookkeeping.Block{
+			BlockHeader: bookkeeping.BlockHeader{
+				Round: basics.Round(i),
+			},
+		}
+		blk.RewardsLevel = rewardLevel
+		blk.CurrentProtocol = consensusVersion
+
+		delta := ledgercore.MakeStateDelta(&blk.BlockHeader, 0, updates.Len(), 0)
+		delta.Accts.MergeAccounts(updates)
+		delta.Creatables = creatablesFromUpdates(base, updates, knownCreatables)
+
+		delta.Totals = accumulateTotals(t, consensusVersion, []map[basics.Address]ledgercore.AccountData{totals}, rewardLevel)
+		allTotals[i] = delta.Totals
+		ml.trackers.newBlock(blk, delta)
+		accts = append(accts, newAccts)
+
+		if i > basics.Round(proto.MaxBalLookback) && i%10 == 0 {
+			onlineTotal, err := ao.OnlineTotals(i - basics.Round(proto.MaxBalLookback))
+			require.NoError(t, err)
+			require.Equal(t, allTotals[i-basics.Round(proto.MaxBalLookback)].Online.Money, onlineTotal)
+			expectedConsensusVersion := protocol.ConsensusCurrentVersion
+			if i > 2*basics.Round(proto.MaxBalLookback) {
```

Comment:
in code above we set future for i > maxballookback:
```		if i > basics.Round(proto.MaxBalLookback) {
			consensusVersion = protocol.ConsensusFuture
		}
```
Why do we expect it to be future only after `2*basics.Round(proto.MaxBalLookback)`?

I think the check should be more strict. We do know that for rounds before `basics.Round(proto.MaxBalLookback` it is current, need to check. and for rounds above `basics.Round(proto.MaxBalLookback` is it feature.
Plus, it would be nice to set some other version as well  for `2*basics.Round(proto.MaxBalLookback)`, so below, after the commit, we can check the db has two different versions.

Reply:
Because we are checking the consensus version at round `i - proto.MaxBalLookback > proto.MaxBalLookback`

Reply:
makes sense, thanks.
I still think more than one protocol switch would be beneficial.